irc-client/main.rkt

```diff
@@ -94,13 +95,18 @@
 (define (irc-send! connection command . args)
   (apply irc:irc-send-command (IrcConnection-internal-connection connection) command args))
 
+(: irc-recv-evt (IrcConnection -> (Evtof IrcMessage)))
+(define (irc-recv-evt connection)
+  (define (handle message)
+    (if (eof-object? message)
+        (error "irc connection closed")
+        (parse-irc-message (cast message irc:irc-message))))
```

Comment:
Is it impossible to get TR to figure out the right type here to avoid the `cast`?

Reply:
I'm not sure, but it didn't like it without the cast. I may experiment more today